src/plugins/obstacle_avoidance_server/obstacle_avoidance_server_impl.cpp

```diff
@@ -0,0 +1,108 @@
+#include "obstacle_avoidance_server_impl.h"
+
+namespace mavsdk {
+
+ObstacleAvoidanceServerImpl::ObstacleAvoidanceServerImpl(System& system) : PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ObstacleAvoidanceServerImpl::ObstacleAvoidanceServerImpl(std::shared_ptr<System> system) :
+    PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ObstacleAvoidanceServerImpl::~ObstacleAvoidanceServerImpl()
+{
+    _parent->unregister_plugin(this);
+}
+
+void ObstacleAvoidanceServerImpl::init()
+{
+    using namespace std::placeholders;
+
+    _parent->register_mavlink_command_handler(
+        MAV_CMD_COMPONENT_CONTROL,
+        std::bind(&ObstacleAvoidanceServerImpl::process_component_control_command, this, _1),
+        this);
+}
+
+void ObstacleAvoidanceServerImpl::deinit()
+{
+    _parent->unregister_all_mavlink_command_handlers(this);
+}
+
+void ObstacleAvoidanceServerImpl::enable() {}
+
+void ObstacleAvoidanceServerImpl::disable() {}
+
+mavlink_message_t ObstacleAvoidanceServerImpl::process_component_control_command(
+    const MavlinkCommandReceiver::CommandLong& command)
+{
+    mavlink_message_t command_ack;
+
+    if (command.params.param1 == MAV_COMP_ID_OBSTACLE_AVOIDANCE) {
+        ObstacleAvoidanceServer::ControlType control_type;
+        control_type.control_type =
+            static_cast<ObstacleAvoidanceServer::ControlType::Type>(command.params.param2);
```

Comment:
```suggestion
            static_cast<ObstacleAvoidanceServer::ControlType::Type>(command.params.param2 + 0.5f);
```
That should be safer for casting, I think.

Reply:
yes for sure